modules/geo-layers/src/quadkey-layer/quadkey-layer.ts

```diff
@@ -0,0 +1,39 @@
+import {AccessorFunction} from '@deck.gl/core';
+import GeoCellLayer from '../geo-cell-layer/GeoCellLayer';
+import {getQuadkeyPolygon} from './quadkey-utils';
+
+const defaultProps = {
+  getQuadkey: {type: 'accessor', value: d => d.quadkey}
+};
+
+/**
+ * Properties of `QuadkeyLayer`.
+ */
+type QuadkeyLayerProps<DataT = any> = {
+  /**
+   * Called for each data object to retrieve the quadkey string identifier.
+   *
+   * By default, it reads `quadkey` property of data object.
+   */
+  getQuadkey?: AccessorFunction<DataT, string>;
+};
+
+export default class QuadkeyLayer<DataT = any, ExtraProps = {}> extends GeoCellLayer<
+  DataT,
+  Required<QuadkeyLayerProps> & ExtraProps
+> {
+  static layerName = 'QuadkeyLayer';
+  static defaultProps: any = defaultProps;
+
+  indexToBounds(): Partial<GeoCellLayer['props']> | null {
+    const {data, getQuadkey} = this.props;
+
+    return {
+      data,
```

Comment:
Not related to this PR, why do you need to pass data here?

Reply:
That's probably question to @felixpalmer 

It appears that while extracting `GeoCellLayer` in #6678 it was useful to leave `data` as responsibility of concrete class: `H3ClusterLayer` needs to pre-process data and keep it in `state` and `QuadKeyLayer` just forward data as is.